test/Tests/src/Data/Range_Spec.enso

```diff
@@ -3,10 +3,34 @@ from Standard.Base import all
 import Standard.Test
 
 spec = Test.group "Range" <|
-    Test.specify "should be created with a start and an end" <|
+    Test.specify "should be created with a start, an end and step" <|
         range = 1.up_to 100
         range.start . should_equal 1
         range.end . should_equal 100
+        range.step  . should_equal 1
+
+        range_2 = 0.up_to 10 . with_step 2
+        range_2.start . should_equal 0
+        range_2.end . should_equal 10
+        range_2.step  . should_equal 2
+
+        range_3 = 2.down_to 0
+        range_3.start . should_equal 2
+        range_3.end . should_equal 0
+        range_3.step  . should_equal -1
+
+    Test.specify "should allow setting a new step magnitude" <|
+        1.up_to 2 . with_step 3 . should_equal (Range 1 2 3)
+
+        0.up_to 10 . with_step 2 . should_equal (Range 0 10 2)
+        0.up_to 10 . with_step 2 . to_vector . should_equal [0, 2, 4, 6, 8]
+
+        10.down_to 0 . with_step 2 . should_equal (Range 10 0 -2)
+        10.down_to 0 . with_step 2 . to_vector . should_equal [10, 8, 6, 4, 2]
+
+        1.up_to 2 . with_step 0.5 . should_fail_with Illegal_Argument_Error
+        1.up_to 2 . with_step -1 . should_fail_with Illegal_Argument_Error
+
     Test.specify "should have a length" <|
         range = 0.up_to 100
         range.length . should_equal 100
```

Comment:
can we have a `down_to` length test and one with non-1 step.

Reply:
These are tested as part of later tests where I check almost all properties of various ranges. Just created not using down_to but Range constructor. So slightly redundant but I guess no harm in adding a few here.

Reply:
Oh - no I actually forgot about the length testing! Sorry and thanks for the catch!